### PR TITLE
Support specializing generic entrypoints in `findAndCheckEntryPoint`.

### LINKED ISSUE #5898: findAndCheckEntryPoint should accept a full declref string.
We should allow the user to compile a specialized entrypoint with `findAndCheckEntryPoint("vertMain<X>")`.

### PR DESCRIPTION
**Pull request by csyonghe**
Closes #5898.

### CODE REVIEW

**Comment by cheneym2** [comment]
Which slang issue does this address?

**Comment by csyonghe** [comment]
@cheneym2 created an issue and linked with this PR.

**Comment by cheneym2** [comment]
Seems like a nice feature.  I don't see what the GLSL test is doing related to the new feature.

File: tools/slang-unit-test/unit-test-glsl-compile.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    auto module = session->loadModuleFromSourceString(
+        "m",
+        "m.slang",
+        userSourceBody,
+        diagnosticBlob.writeRef());
+    SLANG_CHECK(module != nullptr);
+
+    ComPtr<slang::IEntryPoint> entryPoint;
+    module->findAndCheckEntryPoint(
+        "main",
```
Comment (cheneym2) [inline tools/slang-unit-test/unit-test-glsl-compile.cpp:49]: Test doesn't exercise the new feature, is it for something else?

**Comment by csyonghe** [comment]
That test just adds more coverage to the same findAndCheckEnteyPoint API.

**Review by cheneym2 — APPROVED** [review]

File: tools/slang-unit-test/unit-test-glsl-compile.cpp (hunk 1 above)
Comment (csyonghe) [reply tools/slang-unit-test/unit-test-glsl-compile.cpp:49]: No, it doesn’t, but just to exercise the same API a bit more.

**Pull request merged**